### DISCUSSION TITLE
Using html2canvas in a webview

### DISCUSSION THREAD

**Discussion by FredLL-Avaiga** [microsoft/vscode-discussions#210] [Extension Development QnA] [answered]
Hi,

I'm trying to use html2canvas in a webview. It's working fine except for the css files I link into the webpage.
I get an error in the log: 
GET https://file+.vscode-resource.vscode-cdn.net/c%3A/Users.../%40vscode/codicons/dist/codicon.css net::ERR_NAME_NOT_RESOLVED
I don't understand why the "+" is decoded while the ":" or the "@" are not But I'm not sure it's the cause of the pb ...

I tried to use the mediaservices API but it is not available in electron ...

I must say it doesn't prevent the image to be drawn but I don't get the icons :-(

**Comment by isidorn** [comment]
I am not sure if this is supported. @mjbvz might know

**Comment by mjbvz** [comment]
Can you provide a minimal, self-contained example of the issue?

**Comment by FredLL-Avaiga** [comment, marked as answer]
Hi @mjvbz,
I have created a self-contained example of the issue.
I see the same errors in the console that I see in my extension but ... it works (ie the generated image does show the icons)
I will continue to investigate and will let you know the outcome of it ...
Sorry for the noise.

**Reply by FredLL-Avaiga** [reply]
For the record, the issue was linked to HTML icons inside some SVG that was then put inside a foreignObject Tag and thus the icon font needed to be fetched again.
I worked around this by using the dom-to-image-more package instead of html2canvas and specifying the CSP directive connect-src to ${webview.cspSource}.

**Comment by mjvbz** [comment]
> Hi @mjvbz,
> I have created a self-contained example of the issue.
> I see the same errors in the console that I see in my extension but ... it works (ie the generated image does show the icons)
> I will continue to investigate and will let you know the outcome of it ...
> Sorry for the noise.

I am sorry i was not on this for some months... i think it should fix the problems..
